docs/DesignDocs/SGX_QuoteEx_Integration.md

```diff
@@ -0,0 +1,127 @@
+# Attestation: OE SDK Integration with Intel® SGX SDK quote-ex Library for Generation of Evidence in New Formats
+
+This design document proposes an extension of the OE SDK implementation for integration with the Intel® SGX SDK quote-ex library, for support of generation of evidence in new SGX formats such as Enhanced Privacy ID (EPID).
+
+# Motivation
+
+The existing implementation of OE SDK SGX attestation, based on the Intel® SGX SDK Data Center Attestation Primitives (DCAP) quote generation library (simply called the DCAP library), only supports generation of evidence in a single SGX ECDSA-p256 format.
+
+On some SGX platforms, other evidence formats, including those based on the Enhanced Privacy ID (EPID) algorithm, are supported and preferred by some application solutions. Generation of evidence in these formats is supported by the Intel® SGX SDK with a library package libsgx-quote-ex (or simply called quote-ex).
+
+# User Experience
+
+The proposed extension only changes the internal implementation of the OE SDK attestation software stack. It does not impact the [OE SDK attestation API](https://github.com/openenclave/openenclave/blob/master/docs/DesignDocs/CustomAttestation_V3.md). With the integration of the quote-ex library, an attester application enclave's call to OE SDK API `oe_get_attester_plugins()` returns the list of all the SGX evidence attester plugins available to the calling enclave instance.
+
+Integration of the quote-ex library depends on the installation of the Intel® SGX SDK quote-ex library package and its dependencies, as well as proper configuration of the components and their access to dependent backend services. Details for the quote-ex library installation and configuration are outside the scope of this document.
+
+# Specification
+
+## Existing OE SDK Implementation
+
+### Evidence Format Enumeration and Plugin Registration
+
+Existing OE SDK implementation based on the DCAP library only supports generation of evidence in a single SGX ECDSA-p256 format, so there is no need for enumeration of supported evidence formats. As implemented in code file `enclave/sgx/attester.c`, a single attester plugin is created for the SGX ECDSA-p256 evidence format.
+- Note: in the current OE SDK implementation, the UUID for the ECDSA-p256 evidence format is still called `OE_SGX_PLUGIN_UUID`, which is the same as `OE_SGX_ECDSA_P256_PLUGIN_UUID`.
+
+### Implementation of OE SDK API `oe_get_evidence()`
+
+The implementation of OE SDK API `oe_get_evidence()`, in code file `common/attest_plugin.c`, searches for an attester plugin that supports the requested evidence format, and invokes the `get_evidence()` entry point of the selected plugin.
```

Comment:
Suggest: "The implementation" -> "The current implementation" since I think it is still @experimental

Reply:
Agreed.